controller_interface/include/controller_interface/controller_base.h

```diff
@@ -122,12 +126,11 @@ class ControllerBase
   /** \name Non Real-Time Safe Functions
    *\{*/
 
-  /// Get the name of this controller's hardware interface type
-  virtual std::string getHardwareInterfaceType() const = 0;
+  typedef std::vector<hardware_interface::InterfaceResources> ClaimedResourcesType;
```

Comment:
Why call this a "Type" - its a list of resources (and associated hardware interfaces). Maybe just call it ClaimedResources?